homework-classes/RepoView.js

```diff
@@ -18,9 +18,40 @@
      * Renders the repository details.
      * @param {Object} repo A repository object.
      */
+
     render(repo) {
-      // TODO: replace this comment and the console.log with your own code
       console.log('RepoView', repo);
+      const repoContainer = document.querySelector('.repo-container');
+      const select = document.querySelector('.repo-select');
+
+      function addRow(table, title, repoText) {
+        const tr = createAndAppend('tr', table);
+        createAndAppend('th', tr, {
+          text: title,
+        });
+        createAndAppend('td', tr, { text: repoText });
+        return tr;
+      }
+
+      function renderRepoDetails(repo, parent) {
+        const date = new Date(repo.updated_at);
+        const table = createAndAppend('table', parent);
+        const firstRow = addRow(table, 'Repository:', '');
+        createAndAppend('a', firstRow.lastChild, {
+          href: repo.html_url,
+          text: repo.name,
+          target: '_blank',
+        });
+
+        addRow(table, 'Description:', repo.description, 'td');
+        addRow(table, 'Forks:', repo.forks, 'td');
+        addRow(table, 'Updated:', date.toLocaleDateString(), 'td');
+      }
+      renderRepoDetails(repo, repoContainer);
+
+      select.addEventListener('change', () => {
+        repoContainer.innerHTML = '';
```

Comment:
You don't need to have this code inside an event listener. Just erase the previous content each time the `render` method is called by placing this line at the top of that function:

```js
this.container.innerHTML = '';
```